02-express-tutorial/app.js

```diff
@@ -1 +1,72 @@
-console.log('Express Tutorial')
+console.log("Express Tutorial");
+
+const express = require("express");
+const { products } = require("./data");
+
+const app = express(); //  npm start -> http://localhost:3000
+app.use(express.static("./public"));
+
+app.get("/api/v1/test", (req, res) => { // http://localhost:3000/api/v1/test
+    res.json({ msg: "It worked!" });
+});
+
+app.get("/api/v1/products", (req, res) => { // http://localhost:3000/api/v1/products
+    res.json(products);
+});
+
+app.get("/api/v1/products/:productID", (req, res) => {
+    const idToFind = parseInt(req.params.productID);
```

Comment:
There is a chance that `parseInt` fails.  If not handled, your server may crash :)

Reply:
One way you can do this is by wrapping this command in a try..catch block

Reference: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/try...catch

Reply:
Style perference: General best practice is to perfer the guard clause style.

Reference: https://www.geeksforgeeks.org/guard-clause-in-javascript/

Reply:
e.g.
```javascript
if (!product) {
  res.status(404).json({ message: "Product was not found." });
  return
}

res.json(product)
```